pkg/ddevapp/config.go

```diff
@@ -442,7 +447,7 @@ func (app *DdevApp) ValidateConfig() error {
 		v, err := semver.NewVersion(versionconstants.DdevVersion)
 		if err == nil {
 			if !c.Check(v) {
-				return fmt.Errorf("ddev version %v is not supported by this project (%v), use a supported version or update your project config.yaml's `ddev_version_constraint`", versionconstants.DdevVersion, c)
+				return fmt.Errorf("ddev version %v is not supported by this project (%v), use a supported version or update your project config.yaml's `ddev_version_constraint`", versionconstants.DdevVersion, app.DdevVersionConstraint)
```

Comment:
We need to improve this. I was pretty confused by

```
$ ddev config --ddev-version-constraint=">=v1.22.4"
You are reconfiguring the project at /Users/rfay/workspace/d10.
The existing configuration will be updated and replaced.
Configuring a 'drupal10' codebase with docroot 'web' at /Users/rfay/workspace/d10/web
failed to validate config: ddev version v1.22.3-35-g4e684a83a is not supported by this project (>=v1.22.4), use a supported version or update your project config.yaml's `ddev_version_constraint`
```

```suggestion
				return fmt.Errorf("This project has a DDEV version constraint of '%s' and the version of DDEV you are using ('%s') does not meet the constraint. Please update the `ddev_version_constraint` in your .ddev/config.yaml or use a version of DDEV that meets the constraint.", app.DdevVersionConstraint, 
 versionconstants.DdevVersion)
```